vault/resource_pki_secret_backend_crl_config.go

```diff
@@ -183,35 +155,59 @@ func pkiSecretBackendCrlConfigRead(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		log.Printf("[WARN] Removing path %q its ID is invalid, err=%s", path, err)
 		d.SetId("")
-		return fmt.Errorf("invalid path ID %q: %w", path, err)
+		return diag.Errorf("invalid path ID %q: %v", path, err)
 	}
 
 	if config == nil {
 		d.SetId("")
-		return nil
+		return diag.Errorf("received nil response from %q", path)
 	}
 
-	if _, ok := d.GetOk("backend"); !ok {
+	if _, ok := d.GetOk(consts.FieldBackend); !ok {
 		// ensure that the backend is set on import
-		if err := d.Set("backend", strings.TrimRight(path, crlConfigPathBase)); err != nil {
-			return err
+		if err := d.Set(consts.FieldBackend, strings.TrimRight(path, crlConfigPathBase)); err != nil {
+			return diag.Errorf("failed setting field %s: %v", consts.FieldBackend, err)
 		}
 	}
 
-	if err := util.SetResourceData(d, config.Data); err != nil {
```

Comment:
This is the main fix here, passing in config.Data into util.SetResourceData was causing issues when the Data map contained an unknown field